client/src/main/java/org/apache/celeborn/client/read/WorkerPartitionReader.java

```diff
@@ -107,7 +143,9 @@ public ByteBuf next() throws IOException {
       exception.set(ioe);
       throw ioe;
     }
-    returnedChunks++;
+    synchronized (this) {
+      returnedChunks++;
```

Comment:
There can be concurrent issue here. The returned chunk might be from failed location.